src/Views/MainPage/store/actions.js

```diff
@@ -0,0 +1,172 @@
+import axios from 'axios';
+import {
+  CHANGE_PAGE,
+  ADD_SUCCESS,
+  ADD_STARTED,
+  ADD_FAILURE,
+  GET_RANDOM_FACT_REQUEST,
+  GET_RANDOM_FACT_SUCCESS,
+  GET_RANDOM_FACT_FAILURE,
+  GET_DATE_FACT_REQUEST,
+  GET_DATE_FACT_SUCCESS,
+  GET_DATE_FACT_FAILURE,
+  GET_YEAR_FACT_REQUEST,
+  GET_YEAR_FACT_SUCCESS,
+  GET_YEAR_FACT_FAILURE,
+  GET_MATH_FACT_REQUEST,
+  GET_MATH_FACT_SUCCESS,
+  GET_MATH_FACT_FAILURE,
+} from './types';
+
+export function addStarted() {
+  return { type: ADD_STARTED };
+}
+export function addFailure(error) {
+  return { type: ADD_FAILURE, payload: error };
+}
+
+export function getRandomFactRequest() {
+  return { type: GET_RANDOM_FACT_REQUEST };
+}
+
+export function getRandomFactSuccess(response) {
+  return { type: GET_RANDOM_FACT_SUCCESS, payload: response };
+}
+
+export function getRandomFactFailure(error) {
+  return { type: GET_RANDOM_FACT_FAILURE, payload: error };
+}
+
+export function getRandomFact() {
+  return async (dispatch) => {
+    const apiTest = axios.create({
```

Comment:
Nazwa apiTest jest dosyć średnia 😿 Poza tym proponuję zrobić dwie rzeczy!

1. Zrobić gdzieś metodę, która będzie przyjmowała url (bądź jego część) i tylko jego zmieniać podczas tworzenia akcji, w ten sposób pozbędziesz się dużej ilości zduplikowanego kodu
2. Rzeczy takie jak url API (x-rapidapi-host) lub hasło (x-rapidapi-key) powinny być przetrzymywane poza repozytorium, to są poufne dane. Utwórz plik .env i dodaj go do gitignore. Reszta instrukcji:https://facebook.github.io/create-react-app/docs/adding-custom-environment-variables 
Możesz zrobić plik w `/config`, który będzie eksportować dane z .env